src/gui/AstroCalcDialog.cpp

```diff
@@ -3225,24 +3230,20 @@ localSEparameter localSolarEclipse(double JD,int contact,bool central) {
 	double lat = static_cast<double>(core->getCurrentLocation().latitude);
 	double lon = static_cast<double>(core->getCurrentLocation().longitude);
 	double elevation = static_cast<double>(core->getCurrentLocation().altitude);
-	lat = lat * M_PI_180;
 	lon = -(lon);
 	double L = 0.;
 
+	static SolarSystem* ssystem = GETSTELMODULE(SolarSystem);
+	//PlanetP earth = ssystem->getEarth();
+	Vec4d geocentricCoords = ssystem->getEarth()->getRectangularCoordinates(lon,lat,elevation);
+	const double rc = geocentricCoords[0]/(EARTH_RADIUS/AU); // rhoCosPhiPrime
```

Comment:
not sure where EARTH_RADIUS is set here. preferrably `ssystem->getEarth->getEquatorialRadius()` somewhere?

Reply:
Replaced with getEquatorialRadius().